### PR TITLE
Max retry payment Limit with test

### PR DESCRIPTION
**Pull request by Ashutosh619-sudo**
## Summary by CodeRabbit

- **New Features**
	- Enhanced payment processing logic to improve handling of task logs based on their age.
	- Introduced a new field `is_retired` in bills and expense reports to track payment synchronization states.
  
- **Bug Fixes**
	- Improved robustness of the payment creation process by ensuring accurate status setting for task logs.

- **Tests**
	- Added a new test to validate the behavior of the payment creation function under various task log conditions, enhancing test coverage.

### CODE REVIEW

File: apps/sage_intacct/tasks.py [hunk 1]
```diff
@@ -1091,11 +1093,39 @@ def create_ap_payment(workspace_id):
         expense_group__fund_source='PERSONAL'
     ).all()
 
+    print('bills length', len(bills))
```
Comment (ruuushhh) [inline apps/sage_intacct/tasks.py:1096]: remove loggers

File: apps/sage_intacct/tasks.py [hunk 2]
```diff
@@ -1105,6 +1135,7 @@ def create_ap_payment(workspace_id):
                     }
                 )
 
+
```
Comment (ruuushhh) [inline apps/sage_intacct/tasks.py:1138]: remove extra line

File: apps/sage_intacct/tasks.py [hunk 3]
```diff
@@ -1091,11 +1093,39 @@ def create_ap_payment(workspace_id):
...
+                            task_log.save()
+                            continue
+                    
+                    # If created is within 1 month
+                    elif now - relativedelta(months=1) < task_log.created_at:
+                        # Skip if updated within the last week
+                        if task_log.updated_at > now - relativedelta(weeks=1):
+                            task_log.status = 'FAILED'
+                            task_log.save()
+                            continue
```
Comment (ruuushhh) [inline apps/sage_intacct/tasks.py:1127]: move this whole thing to function

**Review by ruuushhh — APPROVED** [review]

File: tests/test_sageintacct/test_tasks.py [hunk 4]
```diff
@@ -1422,3 +1425,94 @@ def test__validate_employee_mapping(mocker, db):
...
+    assert task_log.status == 'FAILED'
+
+    # Update created_at to between 1 and 2 months ago (between 30 and 60 days)
+    TaskLog.objects.filter(task_id='PAYMENT_{}'.format(bill.expense_group.id)).update(
+        created_at=now - timedelta(days=45),  # Between 1 and 2 months ago
+        updated_at=now - timedelta(days=25)  # Updated within the last 1 month
+    )
+    create_ap_payment(workspace_id)
+    task_log.refresh_from_db()
+    assert task_log.status == 'FAILED'
```
Comment (ashwin1111) [inline tests/test_sageintacct/test_tasks.py:1509]: not right metric to assert, we should check updated_at - if it's not updated then it means we skipped

File: apps/sage_intacct/tasks.py [hunk 5]
```diff
@@ -1083,6 +1083,27 @@ def check_expenses_reimbursement_status(expenses, workspace_id, platform, filter
     return is_paid
 
 
+def validate_for_skipping_payment(expense_group_id, workspace_id):
+    task_log = TaskLog.objects.filter(task_id='PAYMENT_{}'.format(expense_group_id), workspace_id=workspace_id, type='CREATING_AP_PAYMENT').first()
```
Comment (ashwin1111) [inline apps/sage_intacct/tasks.py:1087]: type would be diff for ap payment and reimbursement

File: apps/sage_intacct/tasks.py [hunk 6]
```diff
@@ -1093,38 +1114,17 @@ def create_ap_payment(workspace_id):
         expense_group__fund_source='PERSONAL'
```
Comment (ashwin1111) [inline apps/sage_intacct/tasks.py:1114]: is_retired=False

File: apps/sage_intacct/tasks.py [hunk 7]
```diff
@@ -1250,6 +1272,13 @@ def create_sage_intacct_reimbursement(workspace_id):
         expense_group_reimbursement_status = check_expenses_reimbursement_status(
             expense_report.expense_group.expenses.all(), workspace_id=workspace_id, platform=platform, filter_credit_expenses=filter_credit_expenses)
         if expense_group_reimbursement_status:
+
```
Comment (ashwin1111) [inline apps/sage_intacct/tasks.py:1275]: same here, in 1268 - is_retired=False

**Review by ashwin1111 — APPROVED** [review]

**Review by ashwin1111 — APPROVED** [review]

**Pull request merged**